Sources/MapboxMaps/Location/InterpolatedLocationProducer.swift

```diff
@@ -0,0 +1,89 @@
+import CoreLocation
+
+internal protocol InterpolatedLocationProducerProtocol: AnyObject {
+    var location: InterpolatedLocation? { get }
+    func observe(with handler: @escaping (InterpolatedLocation) -> Bool) -> Cancelable
+}
+
+internal final class InterpolatedLocationProducer: NSObject, InterpolatedLocationProducerProtocol {
+    private var startDate: Date?
+    private var endDate: Date?
+    private var startLocation: InterpolatedLocation?
+    private var endLocation: InterpolatedLocation?
+
+    private let observableInterpolatedLocation: ObservableInterpolatedLocationProtocol
+    private let locationInterpolator: LocationInterpolatorProtocol
+    private let dateProvider: DateProvider
+
+    internal var location: InterpolatedLocation? {
+        observableInterpolatedLocation.value
+    }
```

Comment:
@mapbox/maps-ios Is it okay, that this `location` is `nil` until the first call of `participate()` method even if `startLocation` and `endLocation` is set? It prevents configuration of Puck3D here: https://github.com/mapbox/mapbox-maps-ios/blob/8cf3f937cfd0df60fe741ce83a089296c85b2546/Sources/MapboxMaps/Location/Puck/Puck3D.swift#L56

Reply:
Puck3D will be configured as soon as participate is called (on the next display link) because it observes the interpolated location producer: https://github.com/mapbox/mapbox-maps-ios/blob/8cf3f937cfd0df60fe741ce83a089296c85b2546/Sources/MapboxMaps/Location/Puck/Puck3D.swift#L11-L16

Are you running into an issue with this behavior?

Reply:
Yes, we have a test in Nav SDK that verifies [immediate creation of the layer for a 3D puck](https://github.com/mapbox/mapbox-navigation-ios/blob/9e42a853023d9ff1c0ed5535fd693a1b8ceb8411/Tests/MapboxNavigationTests/NavigationViewControllerTests.swift#L359-L372). Do you say this new behavior is intended? In this case we should stop testing this or update the test to work in async way.

Reply:
The new behavior is intended.